src/main/java/com/google/codeu/servlets/HomeServlet.java

```diff
@@ -0,0 +1,42 @@
+package com.google.codeu.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/")
+public class HomeServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws IOException, ServletException {
+
+    // If page does not exist
+    if (request.getServletPath() != "/") {
+      System.err.println("Invalid path requested:");
+      System.err.println("\tpath " + request.getServletPath());
+      System.err.println("\turl " + request.getRequestURL());
+      response.sendError(HttpServletResponse.SC_NOT_FOUND);
+      response.setContentType("text/html;");
+      response.getOutputStream().println("<a href=\"/\">Home</a>");
```

Comment:
If you really want this to be html, then you should make this proper minimal html "<!doctype html><html><body><a href=\"/\">Home</a></body></html>". Another idea is that you could create an error.jsp and put the html in there. Then you could forward the request like you do below.

Also please remove the ';' from the content type string on the line above. It shouldn't be needed.